internal/jimm/cloud.go

```diff
@@ -213,15 +213,43 @@ func (j *JIMM) AddHostedCloud(ctx context.Context, user *openfga.User, tag names
 	if cloud.HostCloudRegion == "" {
 		return errors.E(op, errors.CodeCloudRegionRequired, "cloud host region not specified")
 	}
-	parts := strings.SplitN(cloud.HostCloudRegion, "/", 2)
-	if len(parts) != 2 || parts[0] == "" {
-		return errors.E(op, errors.CodeBadRequest, fmt.Sprintf("invalid cloud/region format %q", cloud.HostCloudRegion))
-	}
-	region, err := j.Database.FindRegion(ctx, parts[0], parts[1])
-	if errors.ErrorCode(err) == errors.CodeNotFound {
-		return errors.E(op, err, errors.CodeNotFound, fmt.Sprintf("unable to find cloud/region %q", cloud.HostCloudRegion))
-	} else if err != nil {
-		return errors.E(op, err)
+
+	var region *dbmodel.CloudRegion
+	var err error
+	if !strings.Contains(cloud.HostCloudRegion, "/") {
+		cl := dbmodel.Cloud{
+			Name: cloud.HostCloudRegion,
+		}
+		if err := j.Database.GetCloud(ctx, &cl); err != nil {
+			return errors.E(op, errors.CodeBadRequest, "unable to find cloud %q", cloud.HostCloudRegion)
```

Comment:
in the godoc of this method there is `If the requested cloud cannot be created on this JAAS system an error with a
 code of CodeIncompatibleClouds will be returned`.
 Maybe there was a reason for it (comply with some juju cli paradigm).
Either change the error code or the godoc.